### ISSUE TITLE
Enable summary for the post in blog

### ISSUE THREAD

**Issue by aldrineeinsteen** [jakewies/hugo-theme-codex#73]
Enable summary into the posts of every blog

![image](https://user-images.githubusercontent.com/4903148/87319127-da5afc80-c546-11ea-8d33-73c26549b59c.png)

The blog post looks odd without any summary.

**Comment by kentnek** [comment]
oh yes I was thinking about this too. I don't mind adding an option to expose the summary, let me try that.

**Comment by jakewies** [comment]
Hey @aldrineeinsteen ! Thanks for your interest in this theme. Hugo themes are overridable, meaning Hugo will look at your project files to resolve a layout before it looks into the theme's files. 

If you installed the theme following the [`README`](https://github.com/jakewies/hugo-theme-codex/blob/master/README.md#getting-started), you should have installed it at `/themes/hugo-theme-codex`. The page you are referring to is the blog list layout, which can be found at `/themes/hugo-theme-codex/layouts/_default/list.html`. If you wish to add a summary to each blog post,  you can copy the contents of `/themes/hugo-theme-codex/layouts/_default/list.html` to your `/layouts/_default/list.html` or `/layouts/blog/list.html` if you are already using the _default list template.

**Comment by kentnek** [comment]
@jakewies oh yes it should be configurable, as I don't want to force that on other people either 😂 I actually had that in mind earlier as some of my blog posts already have the `<--more-->` summary divider, which is supported by the previous theme that I used.

**Comment by jakewies** [comment]
Good idea. How do you envision the configuration for this looking? I think we could separate the `config.toml` params by page. For example:

```toml
# config.toml

[params]
   # ... site params

   # ... blog params
   [blog]
     showSummary = true
```

This is just the first idea that came to my noggin. What do you think? Do you have another solution?

**Comment by jakewies** [comment]
Also, @aldrineeinsteen are you looking to have each blog post's description in the list template? The description of a blog post is defined in each post's `frontmatter.description`. Currently this value is being used as site metadata.

**Comment by kentnek** [comment]
@jakewies no he wanted to show Hugo's [Content Summaries](https://gohugo.io/content-management/summaries/), which is defined in `.Summary`.

I actually wanted to make `showSummary` a site-wide param. Dealing with page-wide params is kinda tricky as the page's key ("blog") is not the same with every user's website. Also a `showSummary` in every page param complicates the config I think, if the user wants to show summaries in all pages.

**Comment by jakewies** [comment]
@kentnek ok cool I didn't know this was an option. I just tested it out, and the [Rich Content example post](https://github.com/jakewies/hugo-theme-codex/blob/master/exampleSite/content/blog/rich-content.md) outputs some wonky text because it uses shortcodes. From first glance it looks like the `.Summary` value and shortcodes don't play nice together.

**Comment by kentnek** [comment]
Hmm then I guess we can add in the `<!--more-->` divider before the rich content for the demo to work.

**Comment by aldrineeinsteen** [comment]
> 
> 
> Good idea. How do you envision the configuration for this looking? I think we could separate the `config.toml` params by page. For example:
> 
> ```toml
> # config.toml
> 
> [params]
>    # ... site params
> 
>    # ... blog params
>    [blog]
>      showSummary = true
> ```
> 
> This is just the first idea that came to my noggin. What do you think? Do you have another solution?

I haven't thought of exposing the same as a configuration; i have pushed a code to display the summary by default along with pagination:

![image](https://user-images.githubusercontent.com/4903148/87338730-3df32300-c563-11ea-9263-34905f6600b0.png)

**Comment by aldrineeinsteen** [comment]
> 
> 
> Hmm then I guess we can add in the `<!--more-->` divider before the rich content for the demo to work.

Yes, when started adding summary found the pagination was missing, i have added the same in #75

**Comment by aldrineeinsteen** [comment]
> 
> 
> Good idea. How do you envision the configuration for this looking? I think we could separate the `config.toml` params by page. For example:
> 
> ```toml
> # config.toml
> 
> [params]
>    # ... site params
> 
>    # ... blog params
>    [blog]
>      showSummary = true
> ```
> 
> This is just the first idea that came to my noggin. What do you think? Do you have another solution?

Added support for configuration in #75
